logstash-core/lib/logstash/plugins/builtin/pipeline/input.rb

```diff
@@ -68,9 +68,11 @@ def internalReceive(events)
         stream_position = stream_position + 1
       end)
       ReceiveResponse.completed()
-    rescue java.lang.InterruptedException, IOError => e
-      # maybe an IOException in enqueueing
-      logger.debug? && logger.debug('queueing event failed', message: e.message, exception: e.class, backtrace: e.backtrace)
+    rescue org.logstash.ackedqueue.QueueRuntimeException, java.io.IOException, IOError => e
```

Comment:
review note: some of Java `IOException`s converted to Ruby `IOError` but majority of (`Queue` and `Page` objects) `IOException`s directly come here.